lite/src/types.rs

```diff
@@ -0,0 +1,105 @@
+/// TrustedState stores the latest state trusted by a lite client,
+/// including the last header and the validator set to use to verify
+/// the next header.
+pub struct TrustedState<H, V>
+where
+    H: Header,
+    V: ValidatorSet,
+{
+    pub last_header: H, // height H-1
+    pub validators: V,  // height H
+}
+
+/// Need to do something better here :)
+pub type Height = u64;
+pub type Hash = u64; // TODO
+pub type Time = u64; // TODO
+pub type Bytes = u64; // TODO
+pub type ValID = u64; // TODO
```

Comment:
If I understand correctly, we still want to figure out what these types should be. As the rest of this package is purely traits, should these types should probably be as general as possible, too. (meaning should they also be traits only describing the expected behaviour and no concrete implementation)? 

Here are some suggestions:

- `ValID` could also be a trait with one method `fn id(&self) -> &[u8];` (`as_bytes` is probably more idiomatic) and the size / length of the id  `fn len() -> usize` (currently it's always 20)
- `Height` could also be a similar trait: so far it should only come with one method: `fn value(self) -> u64`
- `Bytes` 🤔 should either be dropped and `sign_bytes` could directly return `&[u8]`, or the name should be different/more specific and called sth like `Signable` or simply `AsBytes` if we do not want to express that this will be signed; It should also have a single method `as_bytes` or `value`
- similar with `Hash`
- `Time` solely seems to be used to determine if a header is expired for that it seem there are two different concepts: that of a duration and that of a point/instant in time; a trait could come with the necessary subset of Instant and Duration of the [std::time](https://doc.rust-lang.org/std/time/index.html) module (?)

Reply:
OK, these can be concrete instead.